users.js

```diff
@@ -1544,6 +1549,12 @@ class User {
 function pruneInactive(threshold) {
 	let now = Date.now();
 	for (const user of users.values()) {
+		let afkTimer = AFK_TIMER;
+		if (user.can('lock') && !user.can('bypassall')) afkTimer = STAFF_AFK_TIMER;
+		if (user.group !== '*' && !user.connections.some(connection => now - connection.lastActiveTime < afkTimer)) {
+			user.popup(`You have been inactive for over ${afkTimer / (60000)} minutes, and have been logged out of your account as a result.`);
+			user.resetName(false);
```

Comment:
Also, I'm not sure I agree with being logged out. I'd rather they just be marked as inactive in some way. Maybe a new rank symbol?

Reply:
How exactly would the rank symbol work?

Reply:
Maybe do that and have client cut the `|AFK` off and make the name grey & show the user is afk when the name is clicked.

Reply:
You still haven't implemented my suggestions...

Reply:
I'd really like to have this (it's a lot less kludgey than my manual `'pre (away)'` hack for my perma-logged in computer) -  @bumbadadabum, were you planning to pick this back up, or would you want me to take on the rest?

I prefer @HoeenCoder's suggestion of hiding the literal `|AFK` in the client. Making the name grey (and maybe changing the sort order so they show up at the bottom?) seems like it would be sufficient and not add clutter. Though I'd be fine with `|AFK` as well if you feel strongly about it @Zarel.

Reply:
I'm okay with hiding the literal `|AFK`, but I'd recommend some sort of tag or symbol to be much clearer and avoid the chorus of "did you get demoted?"

Reply:
By tag, I mean something along the lines of "AFK" in a rounded rectangle, next to the name.

Reply:
I don't know that we need to remove the rank symbol? I think the server just adds `'|AFK'` (`'@pre'` -> `'@pre|AFK'`), but on the client we hide the tag and grey it like: 
<img width="149" alt="Screen Shot 2019-04-02 at 08 24 16" src="https://user-images.githubusercontent.com/117249/55414910-e58c4380-5520-11e9-9e96-e3ecbff8eb7f.png">, and sort it to the bottom. I think changing the sort would allow us to leave the rank on, as hiding the AFK users at the bottom in grey makes it so that users in the room would be less likely to look to them for action. If you click on the name, it would then show `(away)` or `(AFK)`, 'maybe with some rounded rectangle' that I'm too lazy to do for an example:
<img width="298" alt="Screen Shot 2019-04-02 at 08 27 48" src="https://user-images.githubusercontent.com/117249/55415102-4f0c5200-5521-11e9-8f6c-776586458ac2.png">

Reply:
Sure, works for me.